### DISCUSSION TITLE
"git-lfs: command not found" when cloning

### DISCUSSION THREAD

**Discussion by kmaasrud** [Homebrew/discussions#2368] [Writing Formulae/Casks] [answered]
I am trying to make a new formula for [mdzk](https://github.com/mdzk-rs/mdzk). I'm new to Homebrew formulas and Ruby, but this is what I've got right now:

```ruby
class Mdzk < Formula
  homepage "https://mdzk-rs.github.io"
  version "0.4.2"
  license "MPL-2.0"
  url "https://github.com/mdzk-rs/mdzk.git", tag: "0.4.2"
  depends_on "git-lfs" => :build
  depends_on "rust" => :build

  def install
    system "cargo", "install", *std_cargo_args
  end

  test do
    system "mdzk", "init", "test_mdzk"
    assert_predicate testpath/"test_mdzk", :exist?
  end
end
```

As you can see, the cloning depends on Git LFS. I saw in [an issue](https://github.com/Homebrew/homebrew-core/issues/30190) that simply putting `depends_on "git-lfs" => :build` like I've done, will work. However, when I'm running `brew install --build-from-source mdzk`, I get the following error:

```
git-lfs filter-process: git-lfs: command not found
fatal: the remote end hung up unexpectedly
Error: mdzk: Failed to download resource "mdzk"
Failure while executing; `git checkout -f 0.4.2 --` exited with 128. Here's the output:
git-lfs filter-process: git-lfs: command not found
fatal: the remote end hung up unexpectedly
```

So git-lfs was obviously not installed as a dependency. Has anyone got any idea on how I can solve this? Any help is very much appreciated 😄

**Comment by carlocab** [comment, marked as answer]
Based on the latter part of the discussion in Homebrew/homebrew-core#30190, it doesn't look like just adding `depends_on "git-lfs" => :build` will work, unfortunately.

You might need to create your own [custom download strategy](https://docs.brew.sh/Formula-Cookbook#specifying-the-download-strategy-explicitly) to handle clones that need `git-lfs`.

**Reply by kmaasrud** [reply]
Thank you for the help! I found a way to circumvent git cloning, so all is good 😄
